commit 9f0d26f38ae41684aba5025f4f7445ac6e888094 (refs/changes/87/5587/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2019-02-14T10:19:31-08:00 (8 months ago)
    
    TRITON-1230 CommandExecute needs to watch out for sabotage from validation
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/endpoints/servers.js b/lib/endpoints/servers.js
index 4e81cf4..b802add 100644
--- a/lib/endpoints/servers.js
+++ b/lib/endpoints/servers.js
@@ -1228,7 +1228,9 @@ Server.execute = function handlerCommandExecute(req, res, next) {
 
         if (handleWithUr) {
             // Ur's execute functionality
-            if (req.params.hasOwnProperty('timeout')) {
+            if (req.params.hasOwnProperty('timeout') &&
+                req.params.timeout !== undefined) {
+
                 msg = 'timeout only supported when using cn-agent (See: ' +
                     'FEATURE_USE_CNAGENT_COMMAND_EXECUTE)';
                 self.log.error(msg);
diff --git a/package.json b/package.json
index 3f3a6c7..285b84d 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.19.3",
+  "version": "1.19.4",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
